components/camel-file/src/main/java/org/apache/camel/component/file/FileConsumer.java

```diff
@@ -314,14 +317,15 @@ public FileEndpoint getEndpoint() {
 
     @Override
     protected boolean isMatchedHiddenFile(GenericFile<File> file, boolean isDirectory) {
-        if (getEndpoint().isIncludeHiddenFiles()) {
-            if (isDirectory) {
-                // skip hidden folders
-                String name = file.getFileNameOnly();
-                if (name.startsWith(".")) {
-                    return false;
-                }
+        if (isDirectory) {
+            String name = file.getFileNameOnly();
+            if (!name.startsWith(".")) {
+                return true;
             }
+            return getEndpoint().isIncludeHiddenDir() && !".camel".equals(name);
```

Comment:
It could be a good idea to add ".camel" as a constant on FileConstants so its easier to see its a special folder name, and also use. the constant in FileProcessStrategyFactory

Reply:
well,I think so, i will update